specs/eventing/control-plane.md

```diff
@@ -0,0 +1,942 @@
+
+A Channel represents an [Addressable endpoint](#destination-resolution) (i.e. it
+MUST have a `status.address.url` field) which can receive, store, and forward
+events to multiple recipients (Subscriptions). Subscriptions MUST be associated
+with a Channel based on the `spec.channel` field on the Subscription; it is
+expected that the controller for a Channel will also control the associated
+Subscriptions. When the Channel's `Ready` condition is `true`, the Channel MUST
+provide a `status.address.url` which accepts all valid CloudEvents and MUST
+attempt to forward the received events to each associated Subscription whose
+`Ready` condition is `true`. As described in the
+[Subscription Lifecycle](#subscription-lifecycle) section, a Channel MAY forward
+events to an associated Subscription which does not currently have a `true`
+`Ready` condition, including events received by the Channel before the
+`Subscription` was created.
+
+When a Channel is created, its `spec.channelTemplate` field MAY be populated to
+indicate which of several possible Channel implementations to use. It is
+RECOMMENDED to default the `spec.channelTemplate` field on creation if it is
+unpopulated. Once created, the `spec.channelTemplate` field MUST be immutable;
+the Channel MUST be deleted and re-created to change the `spec.channelTemplate`.
+This pattern is chosen to make it clear that changing `spec.channelTemplate` is
+not an atomic operation and that any implementation would be likely to result in
+event loss during the transition.
+
+### Subscription Lifecycle
+
+A Subscription MAY be created before the referenced Channel indicated by its
+`spec.channel` field. The `spec.channel` object reference MAY refer to either a
+`messaging.knative.dev/v1` Channel resource, or another resource which meets the
+`spec.subscribers` and `spec.delivery` required elements in the Channelable duck
+type. The `spec.channel` reference MUST be to an object in the same namespace;
+specifically, the `spec.channel.namespace` field must be unset or the empty
+string. If the referenced `spec.channel` does not currently exist or its `Ready`
+condition is not `true`, then the Subscription's `Ready` condition MUST NOT be
+`true`, and the reason SHOULD indicate that the corresponding Channel is missing
+or not ready.
+
+The Subscription MUST also set the `status.physicalSubscription` URIs by
+resolving the `spec.subscriber`, `spec.reply`, and
+`spec.delivery.deadLetterSink` as described in
+[Destination resolution](#destination-resolution) before setting the `Ready`
+condition to `true`. If any of the addressable fields fails resolution, the
+Subscription MUST set the `Ready` condition to `false`, and at least one
+condition MUST indicate the reason for the error. The Subscription MUST also set
+`status.physicalSubscription` URIs to the empty string if the corresponding
+`spec` reference cannot be resolved.
+
+At least one of `spec.subscriber` and `spec.reply` MUST be set; if only
```

Comment:
See https://github.com/knative/specs/pull/30; Doug pointed out that this would be simpler to require `spec.subscriber` all the time, and Lionel / Ville agreed on Slack.

We can revisit in https://github.com/knative/specs/pull/30 if we think there's a good reason to allow `subscriber` to be unset and `reply` set instead.